site/app/controllers/admin/ReportController.php

```diff
@@ -652,95 +684,209 @@ public function generateCustomization() {
         }
     }
 
+
+    #[Route("/courses/{_semester}/{_course}/reports/build_form", methods: ['POST'])]
+    public function executeBuildForm(): MultiResponse {
+        // Configure json to go into jobs queue
+        $job_json = (object) [];
+        $job_json->job = 'RunAutoRainbowGrades';
+        $job_json->semester = $this->core->getConfig()->getTerm();
+        $job_json->course = $this->core->getConfig()->getCourse();
```

Comment:
Why not just use an array for this?  Something like this would be a lot cleaner:
```suggestion
        $job_json = [
            'job' => 'RunAutoRainbowGrades',
            'semester' => $this->core->getConfig()->getTerm(),
            'course' => $this->core->getConfig()->getCourse(),
        ];
```